### PR TITLE
SPIFFE ID validation audit in new APIs

### PR DESCRIPTION
**Pull request by azdagron**
This PR contains an audit of the new APIs to ensure we are consistently and appropriately validating the incoming SPIFFE IDs.

### CODE REVIEW

File: pkg/server/api/id_test.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+					return
+				}
+				require.NoError(t, err)
+				require.Equal(t, testCase.expectedID, id)
+			})
+		}
+	}
+
+	t.Run("TrustDomainMemberIDFromProto", func(t *testing.T) {
+		runTests(t, api.TrustDomainMemberIDFromProto, []testCase{
```
Comment (amartinezfayo) [inline pkg/server/api/id_test.go:62]: We may include a case with an empty path?

File: pkg/server/api/id.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	if idutil.IsReservedPath(id.Path()) {
+		return fmt.Errorf("%q is not a workload in trust domain %q; path is in the reserved namespace", id, td)
+	}
+	return nil
+}
+
+// UncheckedIDFromProto converts a SPIFFE ID from the given types.SPIFFEID to
+// spiffeid.ID. It does not do any validation outside of ensuring that the ID
+// is well-formed.
+func UncheckedIDFromProto(protoID *types.SPIFFEID) (spiffeid.ID, error) {
```
Comment (amartinezfayo) [inline pkg/server/api/id.go:84]: `UncheckedIDFromProto` is only used externally from `api_test`. I'm wondering if that suggests that this should not be a public function?

Comment (azdagron) [reply pkg/server/api/id.go:84]: Yeah, I'd be good with that. I'll unexport (and probably rename) it.

File: pkg/server/api/id_test.go (hunk 1 above)
Comment (azdagron) [reply pkg/server/api/id_test.go:62]: Sure! I'll add that to the base tests since it should hold true for all ID types.

Comment (azdagron) [reply pkg/server/api/id_test.go:62]: Actually there is distinct messaging, so it will be a case for each type.

**Pull request merged**